src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,156 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev,T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(tail, value, null);
+        if (size == 0) {
+            head = node;
+        } else {
+            tail.next = node;
+        }
+        tail = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexByAdd(index);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        Node<T> node = new Node<>(null, value, null);
+        if (index == 0) {
+            node.next = head;
+            node.prev = null;
+            head.prev = node;
+            head = node;
+        } else {
+            Node<T> currentNode = findNodeByIndex(index);
+            Node<T> newNode = currentNode.prev;
+            node.next = currentNode;
+            node.prev = newNode;
+            newNode.next = node;
+            currentNode.prev = node;
+        }
```

Comment:
If use create new node inner logic block (after if end else) , you code will simply